extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLSubscriptionHandler.java

```diff
@@ -0,0 +1,125 @@
+import io.smallrye.graphql.execution.error.ExecutionErrorsService;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.ServerWebSocket;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Handler that does the execution of GraphQL Requests
+ */
+public class SmallRyeGraphQLSubscriptionHandler extends SmallRyeGraphQLAbstractHandler {
+
+    private static final Jsonb JSONB = JsonbBuilder.create(new JsonbConfig().withNullValues(true));
+
+    private final ExecutionErrorsService executionErrorsService;
+
+    public SmallRyeGraphQLSubscriptionHandler(Config config, CurrentIdentityAssociation currentIdentityAssociation,
+            CurrentVertxRequest currentVertxRequest) {
+        super(currentIdentityAssociation, currentVertxRequest);
+        this.executionErrorsService = new ExecutionErrorsService(config);
+    }
+
+    @Override
+    protected void doHandle(final RoutingContext ctx) {
+        if (ctx.request().headers().contains(HttpHeaders.UPGRADE, HttpHeaders.WEBSOCKET, true)) {
+            ctx.request().toWebSocket(new Handler<AsyncResult<ServerWebSocket>>() {
+                @Override
+                public void handle(AsyncResult<ServerWebSocket> event) {
+                    if (event.succeeded()) {
+                        ServerWebSocket socket = event.result();
+
+                        socket.closeHandler(new Handler<Void>() {
+                            @Override
+                            public void handle(Void event) {
+                                socket.writeTextMessage("Connection closed");
+                            }
+                        });
+                        socket.textMessageHandler(new Handler<String>() {
+                            @Override
+                            public void handle(String message) {
+                                final AtomicReference<Subscription> subscriptionRef = new AtomicReference<>();
+                                JsonObject jsonInput = inputToJsonObject(message);
+                                ExecutionResponse executionResponse = getExecutionService()
+                                        .execute(jsonInput);
+
+                                ExecutionResult executionResult = executionResponse.getExecutionResult();
+                                if (executionResult != null) {
+
+                                    // If there is error on the query, we can not start a subscription
+                                    if (executionResult.getErrors() != null && !executionResult.getErrors().isEmpty()) {
+                                        socket.writeTextMessage(executionResponse.getExecutionResultAsString());
+                                        socket.close();
+                                    } else {
+                                        socket.writeTextMessage("Connection established.... waiting for data");
```

Comment:
I'm not sure we should be sending a plain text message like this, the processing logic on the client side probably expects a stream of JSON documents and will break upon receiving this

Reply:
Yea, good point. We can do the Connection established in the client itself. I'll remove

Reply:
Ok done. Please have a look again. Thanks :)